src/contact.rs

```diff
@@ -582,120 +579,130 @@ impl Contact {
                     let row_authname: String = row.get(4)?;
 
                     Ok((row_id, row_name, row_addr, row_origin, row_authname))
-                },
-            )
-            .await?
-        {
-            let update_name = manual && name != row_name;
-            let update_authname = !manual
-                && name != row_authname
-                && !name.is_empty()
-                && (origin >= row_origin
-                    || origin == Origin::IncomingUnknownFrom
-                    || row_authname.is_empty());
-
-            row_id = u32::try_from(id)?;
-            if origin >= row_origin && addr.as_ref() != row_addr {
-                update_addr = true;
-            }
-            if update_name || update_authname || update_addr || origin > row_origin {
-                let new_name = if update_name {
-                    name.to_string()
-                } else {
-                    row_name
-                };
+                }).optional()?;
```

Comment:
I personally found using `query_row_optional` easier understandable than a function that returns one particular error into `Ok(None)`. Mabye that's just because I'm used to one way more.

In any case, it should be the same everywhere in the codebase: if we go for using `optional()`, we should at some point get rid of `query_row_optional()`.